src/main/java/com/gradle/enterprise/bamboo/config/JsonConfigurationConverter.java

```diff
@@ -0,0 +1,33 @@
+package com.gradle.enterprise.bamboo.config;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.jetbrains.annotations.Nullable;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+
+@Component
+public class JsonConfigurationConverter {
```

Comment:
You can probably make this component generic by parametrizing `fromJson` method. Don't know if it really needed, but it's cheap and makes naming a bit simpler. 

```
public class JsonConverter {

    @Nullable
    public <T> T fromJson(@Nullable String json, Class<T> type) {
        ....
    }
}
```

Reply:
I would also consider making it a util class, as we don't have any state here (we also don't mock it in tests)